annotation-web/src/main/java/eu/europeana/annotation/web/validation/BaseAnnotationValidator.java

```diff
@@ -524,10 +533,38 @@ void validateEdmRights(Body body, Authentication authentication)
       licence = licence.substring(0, versionStart);
     }
     Set<String> rights = getConfiguration().getAcceptedLicenceses();
-    if (!rights.contains(licence))
-      throw new RequestBodyValidationException(body.getInputString(),
-          I18nConstants.INVALID_PARAM_VALUE, new String[] {BODY_EDM_RIGHTS, rightsClaim});
-
+    if (!rights.contains(licence)) {
+    	//if license not supported, check the jwt affiliation
+        EuropeanaApiCredentials apiCred = ((EuropeanaApiCredentials) authentication.getCredentials());
+        if(!validateDataProviderAffiliation(target.getResourceId(), apiCred.getAffiliation())) {    
+            throw new RequestBodyValidationException(body.getInputString(),
+                    I18nConstants.INVALID_PARAM_VALUE, new String[] {BODY_EDM_RIGHTS, rightsClaim});
+        }        
+    }
+  }
+  
+  private boolean validateDataProviderAffiliation(String recordId, String affiliation) {
+	  if(recordId==null || affiliation==null) {
+		  return false;
+	  }
+	  String url = getConfiguration().getSearchApiBaseUrl();
+	  url += "?";
+	  url += "query=europeana_id:" + URLEncoder.encode("\"" + recordId + "\"", StandardCharsets.UTF_8);
+	  url += "&";
+	  url += "qf=foaf_organization:" + URLEncoder.encode("\"" + affiliation + "\"", StandardCharsets.UTF_8);
+	  url += "&";
+	  url += "rows=0";
+	  url += "&";
+	  url += "wskey=apidemo";
+	  try {
+		String searchApiResp = getHttpConnection().getURLContentAsString(url, "Accept", "application/json");
+		@SuppressWarnings("unchecked")
+		Map<String, Object> searchApiRespMap = (Map<String, Object>) JsonUtils.getObjectMapper().readValue(searchApiResp, Map.class);
+		int totalResults = (int) searchApiRespMap.get("totalResults");
```

Comment:
the retrieval of results from SearchApi must be moved to own class SearchApiClient